### DISCUSSION TITLE
Is there fluent-bit release for macOS M2?

### DISCUSSION THREAD

**Discussion by valof** [fluent/fluent-bit#8756] [Q&A] [answered]
Are you planning to release arm64 build for M2?

**Comment by patrick-stephens** [comment]
It's very expensive to build officially for OSS from a CI perspective. I'm not sure if there are any blockers there on the actual build platform we have in Github but fundamentally it is buildable.

Commercially Calyptia does provide multi-arch builds for all targets if you want them and the code should build fine so you can always build it yourself directly on platform.

**Reply by valof** [reply]
Thank you, patrick-stephens! I would like to know if there are any blockers (such as dependencies with no M2 support, etc.)
You are saying "Commercially Calyptia does provide multi-arch builds" - where can I see the price, etc. can you or anyone tell me?

**Reply by patrick-stephens** [reply]
There's no inherent issue as far as I'm aware, obviously it depends on exactly what you want to build but it should be fine.

https://calyptia.com/ is where I'd start, I don't really want to advertise here on an OSS discussion but hopefully that points you in the right direction. Feel free to ping me or @agup006 on the Fluent Bit Slack as well with the link on the docs: https://launchpass.com/fluent-all

**Reply by valof** [reply]
Thank you, patrick-stephens!
I went to the build process, met the following error:
```
[ 15%] Building C object plugins/processor_sql/parser/CMakeFiles/processor-sql-parser.dir/sql_parser.c.o
/Users/ofshtev/git/fluent-bit/plugins/processor_sql/parser/sql_parser.c:12:10: fatal error: 'processor-sql_parser.h' file not found
#include "processor-sql_parser.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~

```
It looks like a genuine error, not something I forgot to specify for the build... now while I am writing it, I realize that I took the master branch... which is probably an indication the problem is still between the chair and keyboard...
I will try an exact release we need - unless you tell me I have to install something for plugins.

**Reply by valof** [reply]
Oh, for some reason I cannot get to Slack - I will do it later.

**Reply by valof** [reply]
Now, when I went to a release 2.2.2 there is a different error, but similar in nature. Maybe I have to use the Bison of a specific version? 
--------------------------------------
[ 22%] [BISON][cmt_decode_prometheus_parser] Building parser with bison 3.8.2
cmt_decode_prometheus.y:2.1-37: warning: deprecated directive: ‘%name-prefix "cmt_decode_prometheus_"’, use ‘%define api.prefix {cmt_decode_prometheus_}’ [-Wdeprecated]
    2 | %name-prefix "cmt_decode_prometheus_"
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      | %define api.prefix {cmt_decode_prometheus_}
cmt_decode_prometheus.y: warning: 5 shift/reduce conflicts [-Wconflicts-sr]
cmt_decode_prometheus.y: note: rerun with option '-Wcounterexamples' to generate conflict counterexamples
xcode-select: Failed to locate 'm4', requesting installation of command line developer tools.
cmt_decode_prometheus.y: warning: fix-its can be applied.  Rerun with option '--update'. [-Wother]
[ 22%] [BISON][cmt_decode_prometheus_parser] Building parser with bison 3.8.2
cmt_decode_prometheus.y:2.1-37: warning: deprecated directive: ‘%name-prefix "cmt_decode_prometheus_"’, use ‘%define api.prefix {cmt_decode_prometheus_}’ [-Wdeprecated]
    2 | %name-prefix "cmt_decode_prometheus_"
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      | %define api.prefix {cmt_decode_prometheus_}
cmt_decode_prometheus.y: warning: 5 shift/reduce conflicts [-Wconflicts-sr]
cmt_decode_prometheus.y: note: rerun with option '-Wcounterexamples' to generate conflict counterexamples
xcode-select: Failed to locate 'm4', requesting installation of command line developer tools.
cmt_decode_prometheus.y: warning: fix-its can be applied.  Rerun with option '--update'. [-Wother]
[ 22%] Building C object lib/cmetrics/src/CMakeFiles/cmetrics-static.dir/__/cmt_decode_prometheus_lexer.c.o
In file included from cmt_decode_prometheus.l:9:
/Users/ofshtev/git/fluent-bit/lib/cmetrics/include/cmetrics/cmt_decode_prometheus.h:95:10: fatal error: 'cmt_decode_prometheus_parser.h' file not found
#include "cmt_decode_prometheus_parser.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make[2]: *** [lib/cmetrics/src/CMakeFiles/cmetrics-static.dir/__/cmt_decode_prometheus_lexer.c.o] Error 1
make[1]: *** [lib/cmetrics/src/CMakeFiles/cmetrics-static.dir/all] Error 2
make: *** [all] Error 2
-----------------------------

**Reply by patrick-stephens** [reply, marked as answer]
I'm guessing this is resolved now per https://fluent-all.slack.com/archives/C0CTQGHKJ/p1714106936072559?thread_ts=1714098596.026579&cid=C0CTQGHKJ :)

**Comment by patrick-stephens** [comment]
We could actually start using the new Github hosted macOS runners to do this if you want to submit a PR to add them to https://github.com/fluent/fluent-bit/blob/master/.github/workflows/call-build-macos.yaml

We can switch it to a matrix job like the Windows one and everything should magically drop out then: https://github.com/fluent/fluent-bit/blob/e47718ad2368240dc5fee2abdc5d520d1ac23f6c/.github/workflows/call-build-windows.yaml#L81

@cosmo0920 might be an idea?

**Reply by cosmo0920** [reply]
Can we use GitHub hosted M1 runners? Then, We have to rewrite `runs-on` part on call-build-macos.yaml
